### PR TITLE
Make snorkel installable as a package

### LINKED ISSUE #920: Make snorkel installable as a package
It would be make snorkel more portable if it were installable as a package. I'm happy to try to tackle this today and tomorrow (while we're at Stanford Info Lab). The plan would be to create a `setup.py` so snorkel could be installed as a package in a versioned way. This hopefully would reduce the need to set environment variables, which has been causing us headaches.

Related content:

+ `treedlib` is now pip installable (thanks to @lukehsiao in https://github.com/HazyResearch/treedlib/pull/43)
+ here's the [conda environment](https://github.com/greenelab/snorkeling/blob/6336bc6769e8c1edae5e4368dbd0c4d4dcd3e7de/environment.yml) we're currently using which could be a good reference.
+ Refs  https://github.com/HazyResearch/snorkel/issues/909
+ Refs https://github.com/HazyResearch/snorkel/pull/918
+ Refs https://github.com/HazyResearch/snorkel/issues/505

@ajratner and @stephenbach: do you know of any reasons that prevent using snorkel as a package rather than submodule?

### PR DESCRIPTION
**Pull request by dhimmel**
Closes #920.
Closes https://github.com/HazyResearch/snorkel/pull/918

Work in progress

- Reformatted `snorkel/features/entity_features.py` which imported treedlib from the submodule.
- Part of Snorkel Week https://github.com/greenelab/snorkeling/issues/39

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,4 +0,0 @@
-[submodule "treedlib"]
```
Comment (stephenbach) [inline .gitmodules]: 💯

File: .travis.yml [hunk 2]
```diff
@@ -54,21 +45,14 @@ before_install:
...
-  # Install Numba
-  - conda install --yes numba
+  - sed --in-place 's/- python/- python='"$PYTHON_VERSION"'/' environment.yml
+  - conda env create --quiet --file=environment.yml
+  - conda activate snorkel
 
   # Install all remaining dependencies as per our README
-  - pip install -r python-package-requirement.txt
+  - pip install .
   - test -e parser/corenlp.sh || ./install-parser.sh
```
Comment (stephenbach) [inline .travis.yml]: Can we get rid of CoreNLP installation?

File: snorkel/viewer.py [hunk 3]
```diff
@@ -16,8 +16,7 @@
 import warnings
 warnings.filterwarnings("ignore", category=DeprecationWarning)
 
-HOME = os.environ['SNORKELHOME']
-
+HOME = os.path.abspath(os.path.join(__file__, '..', '..'))
```
Comment (stephenbach) [inline snorkel/viewer.py]: Need to move .html, etc. files into module?

File: README.md [hunk 4]
```diff
@@ -73,36 +73,64 @@ Then, for more content, check out the other tutorials avaliable [here](https://g
...
-Finally, enable `ipywidgets`:
-```bash
-jupyter nbextension enable --py widgetsnbextension --sys-prefix
+However, installing Snorkel via pip will not install dependencies.
+Instead, to manage its environment, Snorkel uses [conda](http://conda.pydata.org/docs/), which allows specifying dependencies via an `environment.yml` file.
+This documentation covers two common cases (usage and development) for setting up conda environments for Snorkel.
+In both cases, the environment can be activated using `conda activate snorkel` and deactivated using `conda deactivate`.
+With older versions of conda, you may have to replace `conda` with `source` in the above commands.
+
+### Usage Environment
```
Comment (stephenbach) [inline README.md]: First instructions user should hit is how to install Snorkel as a user (dev environment can come lower in instructions).

File: README.md [hunk 5]
```diff
@@ -73,36 +73,64 @@ Then, for more content, check out the other tutorials avaliable [here](https://g
...
+3. Add the `snorkel` package installation to your `environment.yml`, under the `- pip` section.
+Of course, we suggest versioning snorkel, which you can do via a release number or commit hash (to access more bleeding edge functionality)
+  ```yml
+    # Versioned via release tag
+    - git+https://github.com/HazyResearch/snorkel@v0.6.3
+    # Versioned via commit hash (commit hash below is fake to ensure you change it)
+    - git+https://github.com/HazyResearch/snorkel@7eb7076f70078c06bef9752f22acf92fd86e616a
+  ```
+Finally, consider versioning the `numbskull` and `treedlib` pip dependencies by changing `master` to their latest commit hash on GitHub.
+
```
Comment (stephenbach) [inline README.md]: End of install instructions should emphatically point users to tutorials.

File: run.sh [hunk 6]
```diff
@@ -2,9 +2,6 @@
 source set_env.sh
```
Comment (stephenbach) [inline run.sh]: Do we need this and `set_env.sh`? Right now this script installs CoreNLP, which we don't want to do automatically anymore, and launches jupyter.

**Comment by dhimmel** [comment]
@ajratner would be great if you can take a look at this PR!

**Pull request merged**